farmdata2/farmdata2_modules/resources/BannerComponent.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * A Vue component that augments the Bootstrap CSS class for the alert banner to generate dynamic success/error/message banner for pages.
+ * 
+ * <h3 class="subsection-title">data-cy attributes</h3>
+ * <table>
+ * <thead><tr><th>Value</th>        <th>Description</th></tr></thead>
+ * <tbody>
+ * <tr><td>banner-handler</td>   <td>div containing bootstrap alert banner</td></tr>
+ * <tr><td>banner-close</td>   <td>'X' button that closes the alert banner</td></tr>
+ * <tr><td>banner-message</td>   <td>The message displayed inside the alert banner</td></tr>
+ * </tbody>
+ * </table>
+ * 
+ * <div>
+ *  @vue-prop {Object} [message={"msg": "Hello, I am a banner alert.", "class": "alert alert-warning"}] - Contains the message and class for the alert banner
+ *  @vue-prop {Boolean} [visible=null] - Set to true to show the alert banner. False makes the banner invisible.
+ *  @vue-prop {Boolean} [timeout=false] - Sets whether or not the banner dismisses itself which disables the 'x' to close. Set to false by default. Currently set to self-dismiss after 5 seconds.
+ * </div>
+ * 
+ * <div>
+ *  @vue-event visibility-update - Updates the visibility in the parent page when the child component has internal visibility updates. This means each page needs to implement a function that catches this emit and matches their visibility with the child's internal visibility. 
+ *  @returns false because the banner has been dismissed (either via clicking the x or the timeout).
+ *  @example <caption>Retrieve all planting assets into a new array.</caption>
+ * updateBannerVisibility(newBool) {
+ * this.bannerVisibility = newBool 
+ * // For a specific example make sure to check the UI page's source code. 

```

Comment:
I would like to see a code example here that shows how to respond to the `banner-hidden` event.  That piece is something that will be easy to mess up and should be documented.